airbyte-workers/src/main/java/io/airbyte/workers/ApplicationInitializer.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers;
+
+import io.airbyte.analytics.Deployment;
+import io.airbyte.analytics.TrackingClientSingleton;
+import io.airbyte.commons.version.AirbyteVersion;
+import io.airbyte.config.Configs.DeploymentMode;
+import io.airbyte.config.Configs.TrackingStrategy;
+import io.airbyte.config.Configs.WorkerEnvironment;
+import io.airbyte.config.Configs.WorkerPlane;
+import io.airbyte.config.MaxWorkersConfig;
+import io.airbyte.config.helpers.LogClientSingleton;
+import io.airbyte.config.helpers.LogConfigs;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.db.check.DatabaseCheckException;
+import io.airbyte.db.check.DatabaseMigrationCheck;
+import io.airbyte.db.check.impl.JobsDatabaseAvailabilityCheck;
+import io.airbyte.metrics.lib.MetricClientFactory;
+import io.airbyte.metrics.lib.MetricEmittingApps;
+import io.airbyte.scheduler.persistence.JobPersistence;
+import io.airbyte.workers.process.KubePortManagerSingleton;
+import io.airbyte.workers.temporal.TemporalJobType;
+import io.airbyte.workers.temporal.TemporalUtils;
+import io.airbyte.workers.temporal.check.connection.CheckConnectionWorkflowImpl;
+import io.airbyte.workers.temporal.discover.catalog.DiscoverCatalogWorkflowImpl;
+import io.airbyte.workers.temporal.scheduling.ConnectionManagerWorkflowImpl;
+import io.airbyte.workers.temporal.spec.SpecWorkflowImpl;
+import io.airbyte.workers.temporal.support.TemporalProxyHelper;
+import io.airbyte.workers.temporal.sync.SyncWorkflowImpl;
+import io.grpc.StatusRuntimeException;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.annotation.Value;
+import io.micronaut.context.env.Environment;
+import io.micronaut.context.event.ApplicationEventListener;
+import io.micronaut.discovery.event.ServiceReadyEvent;
+import io.micronaut.scheduling.TaskExecutors;
+import io.temporal.api.workflowservice.v1.DescribeNamespaceRequest;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import io.temporal.worker.Worker;
+import io.temporal.worker.WorkerFactory;
+import io.temporal.worker.WorkerOptions;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Performs any required initialization logic on application context start.
+ */
+@Singleton
+@Requires(notEnv = {Environment.TEST})
+@Slf4j
+public class ApplicationInitializer implements ApplicationEventListener<ServiceReadyEvent> {
```

Comment:
This will lead to have a lot of Bean to be initialized for all the project depending on the worker one. It shuld not be added to the review but for the next project being converted to Micronaut, we should extract the dependencies they have on the worker if possible.